backend/app/assets/javascripts/spree/backend/product_picker.js

```diff
@@ -4,6 +4,15 @@ $.fn.productAutocomplete = function (options) {
   // Default options
   options = options || {}
   var multiple = typeof(options['multiple']) !== 'undefined' ? options['multiple'] : true
+  function ransackQuery(term) {
+    if (typeof(options['ransackQCallback']) === 'function') return options['ransackQCallback'](term)
```

Comment:
I really like that the `term` is passed into the function ( a bug we currently have in the `variantAutocomplete` actually)

But could we use a more approachable name for the option and maybe use the same as in the `variantAutocomplete`? https://github.com/solidusio/solidus/blob/master/backend/app/assets/javascripts/spree/backend/variant_autocomplete.js#L11

```suggestion
    if (typeof(options['searchOptions']) === 'function') return options['searchOptions'](term)
```

**Extra bonus points** for fixing the `variantAutocomplete` in this PR as well ;)

Reply:
I think I might move the variant work to another PR @tvdeyen. Also, I am wondering if/how we should change this `variantComplete`. Technically, it one can currently initialize it to pass to the `ransack` method directly, but doesn't have access to the `select2` search term. Also, if doesn't go down this route, it will end up using a search variant class (`Spree::Core::Search::Variant` by default) which does has a lot of search logic already.

Having all these different paths is adding a lot of complexity. If we start accepting a function in the JS instead of a hash, having the search variant class route almost becomes redundant 🤔 
https://github.com/solidusio/solidus/blob/d582f545fb51eeb1c4b4cf318009c0a7f28db774/api/app/controllers/spree/api/variants_controller.rb#L35
https://github.com/solidusio/solidus/blob/d582f545fb51eeb1c4b4cf318009c0a7f28db774/core/lib/spree/core/search/variant.rb